File: results/2026-04-09_152435/tasks/17-artifact-cleanup-script/typescript-bun-opus/generated-code/fixtures.ts
// fixtures.ts — Test fixture data for artifact cleanup tests
// Reference date: 2026-04-10T00:00:00Z
//
// Artifact ages from reference date:
//   deploy-bundle-1:  54 days (Feb 15)
//   build-artifact-1: 40 days (Mar 1)
//   test-results-1:   31 days (Mar 10)
//   build-artifact-2: 26 days (Mar 15)
//   build-artifact-3:  9 days (Apr 1)
//   test-results-2:    5 days (Apr 5)
//   deploy-bundle-2:   2 days (Apr 8)
//
// Total size: 13,750,000 bytes

import type { Artifact } from "./types";

export const REFERENCE_DATE = new Date("2026-04-10T00:00:00Z");

export const TEST_ARTIFACTS: Artifact[] = [
  {
    name: "build-artifact-1",
    sizeBytes: 1_000_000,
    createdAt: "2026-03-01T00:00:00Z",
    workflowRunId: "workflow-a",
  },
  {
    name: "build-artifact-2",
    sizeBytes: 2_000_000,
    createdAt: "2026-03-15T00:00:00Z",
    workflowRunId: "workflow-a",
  },
  {
    name: "build-artifact-3",
    sizeBytes: 1_500_000,
    createdAt: "2026-04-01T00:00:00Z",
    workflowRunId: "workflow-a",
  },
  {
    name: "test-results-1",
    sizeBytes: 500_000,
    createdAt: "2026-03-10T00:00:00Z",
    workflowRunId: "workflow-b",
  },
  {
    name: "test-results-2",
    sizeBytes: 750_000,
    createdAt: "2026-04-05T00:00:00Z",
    workflowRunId: "workflow-b",
  },
  {
    name: "deploy-bundle-1",
    sizeBytes: 5_000_000,
    createdAt: "2026-02-15T00:00:00Z",
    workflowRunId: "workflow-c",
  },
  {
    name: "deploy-bundle-2",
    sizeBytes: 3_000_000,
    createdAt: "2026-04-08T00:00:00Z",
    workflowRunId: "workflow-c",
  },
];

File: results/2026-04-17_004319/tasks/17-artifact-cleanup-script/default-haiku45/generated-code/fixtures.py
"""Test fixtures for cleanup scenarios."""
import json
from datetime import datetime, timedelta


def create_test_case_1():
    """
    Test case 1: Basic cleanup by age.
    3 artifacts, 1 too old.
    """
    now = datetime.now()
    return {
        "artifacts": [
            {
                "name": "build-old.zip",
                "size_bytes": 5000,
                "created_at": (now - timedelta(days=31)).isoformat(),
                "workflow_run_id": "run-1",
            },
            {
                "name": "build-recent.zip",
                "size_bytes": 5000,
                "created_at": (now - timedelta(days=15)).isoformat(),
                "workflow_run_id": "run-1",
            },
            {
                "name": "build-newest.zip",
                "size_bytes": 5000,
                "created_at": now.isoformat(),
                "workflow_run_id": "run-1",
            },
        ]
    }


def create_test_case_2():
    """
    Test case 2: Multiple workflows.
    Keep latest 2 per workflow independently.
    """
    now = datetime.now()
    return {
        "artifacts": [
            {
                "name": "wf1-build-1.zip",
                "size_bytes": 1000,
                "created_at": (now - timedelta(days=4)).isoformat(),
                "workflow_run_id": "workflow-1",
            },
            {
                "name": "wf1-build-2.zip",
                "size_bytes": 1000,
                "created_at": (now - timedelta(days=3)).isoformat(),
                "workflow_run_id": "workflow-1",
            },
            {
                "name": "wf1-build-3.zip",
                "size_bytes": 1000,
                "created_at": (now - timedelta(days=2)).isoformat(),
                "workflow_run_id": "workflow-1",
            },
            {
                "name": "wf2-build-1.zip",
                "size_bytes": 1000,
                "created_at": (now - timedelta(days=3)).isoformat(),
                "workflow_run_id": "workflow-2",
            },
            {
                "name": "wf2-build-2.zip",
                "size_bytes": 1000,
                "created_at": (now - timedelta(days=1)).isoformat(),
                "workflow_run_id": "workflow-2",
            },
        ]
    }


def create_test_case_3():
    """
    Test case 3: Total size exceeded.
    5KB limit, 6KB total, should delete oldest.
    """
    now = datetime.now()
    return {
        "artifacts": [
            {
                "name": "build-1.zip",
                "size_bytes": 2000,
                "created_at": (now - timedelta(days=10)).isoformat(),
                "workflow_run_id": "run-1",
            },
            {
                "name": "build-2.zip",
                "size_bytes": 2000,
                "created_at": (now - timedelta(days=5)).isoformat(),
                "workflow_run_id": "run-1",
            },
            {
                "name": "build-3.zip",
                "size_bytes": 2000,
                "created_at": now.isoformat(),
                "workflow_run_id": "run-1",
            },
        ]
    }


def save_fixture(name: str, data: dict):
    """Save fixture to JSON file."""
    filename = f"fixtures/test_case_{name}.json"
    with open(filename, 'w') as f:
        json.dump(data, f, indent=2)


if __name__ == '__main__':
    import os
    os.makedirs('fixtures', exist_ok=True)

    save_fixture('1_basic_age', create_test_case_1())
    save_fixture('2_multiple_workflows', create_test_case_2())
    save_fixture('3_size_exceeded', create_test_case_3())

    print("✓ Test fixtures created")
